straxen/analyses/event_display.py

```diff
@@ -93,16 +114,18 @@ def event_display(context,
     if records_matrix not in ('raw', True, False):
         raise ValueError('Choose either "raw", True or False for records_matrix')
     if ((records_matrix == 'raw' and not context.is_stored(run_id, 'raw_records')) or
-            (records_matrix and not context.is_stored(run_id, 'records'))):
+        (isinstance(records_matrix, bool) and records_matrix and not context.is_stored(run_id, 'records'))):   # noqa

```

Comment:
You can simplify this here since you check already above that `records_matrix` is either 'raw' or bool. It should be
`bool(raw) == True` Hence `if records_matrix and (...)` should be enough.

Reply:
Yeah, I thought so too, but what happens if we have records_matrix == 'raw' and we don't have 'records' stored. Then `bool(raw) == True and not context.is_stored(run_id, 'records')` so that is condition is True. I also confused myself a little here.

Or perhaps I'm not seeing what you mean. I think strictly speaking:
`` isinstance(records_matrix, bool) and not context.is_stored(run_id, 'records'))`

is indeed all one needs.